src/components/KRadio/KRadio.vue

```diff
@@ -376,51 +370,83 @@ $kRadioDotSize: 6px;
 
       &.has-label, &.has-description {
         .card-content-wrapper {
+          height: auto;
           margin-bottom: var(--kui-space-40, $kui-space-40);
         }
       }
 
       .radio-label {
         @include labelDefaults;
       }
+
+      :deep(#{$kongponentsKongIconSelector}) {
+        color: var(--kui-color-text-neutral, $kui-color-text-neutral) !important;
+      }
     }
 
-    &.checked.radio-card {
+    &.checked.radio-card:not(.disabled) {
       .radio-card-wrapper {
-        background-color: var(--kui-color-background-primary-weakest, $kui-color-background-primary-weakest);
-        box-shadow: var(--kui-shadow-border-primary, $kui-shadow-border-primary);
+        border: var(--kui-border-width-20, $kui-border-width-20) solid var(--kui-color-border-primary, $kui-color-border-primary);
+
+        &:hover {
+          border: var(--kui-border-width-20, $kui-border-width-20) solid var(--kui-color-border-primary-strong, $kui-color-border-primary-strong);
+        }
+
+        .radio-label {
+          color: var(--kui-color-text-primary, $kui-color-text-primary);
+        }
+
+        :deep(#{$kongponentsKongIconSelector}) {
+          color: var(--kui-color-text-primary, $kui-color-text-primary) !important;
+        }
       }
     }
 
-    &.input-error {
+    &.input-error.radio-card:not(.disabled) {
       .radio-card-wrapper {
-        box-shadow: var(--kui-shadow-border-danger, $kui-shadow-border-danger);
+        border: var(--kui-border-width-10, $kui-border-width-10) solid var(--kui-color-border-danger, $kui-color-border-danger);
 
         &:hover {
-          box-shadow: var(--kui-shadow-border-danger-strong, $kui-shadow-border-danger-strong);
+          border: var(--kui-border-width-10, $kui-border-width-10) solid var(--kui-color-border-danger-strong, $kui-color-border-danger-strong);
         }
 
         &:focus-visible {
-          box-shadow: var(--kui-shadow-border-danger-strong, $kui-shadow-border-danger-strong), var(--kui-shadow-focus, $kui-shadow-focus);
+          box-shadow: var(--kui-shadow-focus, $kui-shadow-focus);
         }
       }
 
       &.checked {
         .radio-card-wrapper {
-          background-color: var(--kui-color-background-danger-weakest, $kui-color-background-danger-weakest);
-          box-shadow: var(--kui-shadow-border-danger, $kui-shadow-border-danger);
+          border: var(--kui-border-width-20, $kui-border-width-20) solid var(--kui-color-border-danger, $kui-color-border-danger);
+
+          &:hover {
+            border: var(--kui-border-width-20, $kui-border-width-20) solid var(--kui-color-border-danger-strong, $kui-color-border-danger-strong);
+          }
+
+          .radio-label {
+            color: var(--kui-color-text, $kui-color-text);
+          }
+
+          :deep(#{$kongponentsKongIconSelector}) {
+            color: var(--kui-color-text, $kui-color-text) !important;
+          }
```

Comment:
issue: Shouldn't the host app have control over the icon it slots? There would be no way to override this.

Reply:
Same as above.